lib/node_modules/@stdlib/lapack/blas/dlaswp/lib/dlaswp.js

```diff
@@ -0,0 +1,133 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isLayout = require( '@stdlib/blas/base/assert/is-layout' );
+
+
+// MAIN //
+
+/**
+* Performs a series of row interchanges on the matrix `A`.
+*
+* @param {PositiveInteger} N - number of columns of `A`
+* @param {Array} A - matrix
+* @param {PositiveInteger} LDA - leading dimension of `A`
+* @param {PositiveInteger} k1 - first element of `IPIV` for which a row interchange will be done
+* @param {PositiveInteger} k2 - second element of `IPIV` for which a row interchange will be done
+* @param {Array} IPIV - vector of pivot indices
+* @param {PositiveInteger} incX - increment between successive values of `IPIV`
+* @returns {Float64Array} permuted matrix `A`
+*
+* @example
+* TODO:
+*/
+function dlaswp( order, N, A, LDA, k1, k2, IPIV, incX ) {
```

Comment:
@kgryte I will request you to take a look at implementation, if this seems fine, I'll add remaining files.

Reply:
When I use this example to test it:

```javascript
var Float64Array = require( '@stdlib/array/float64' );
var Int32Array = require( '@stdlib/array/int32' );
var dlaswp = require( './../lib' );

var IPIV = new Int32Array( [ 3, 0, 1, 2 ] );
var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0 ] );

var out = dlaswp( 'column-major', 4, A, 4, 1, 4, IPIV, 1 );

console.log( out );

IPIV = new Int32Array( [ 2, 0, 1 ] );
A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );

out = dlaswp( 'column-major', 3, A, 2, 1, 3, IPIV, 1 );

console.log( out );
```

```console
% node /Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/lapack/blas/dlaswp/examples/index.js
Float64Array(16) [
   2,  3,  1,  4, 6,  7,
   5,  8, 10, 11, 9, 12,
  14, 15, 13, 16
]
Float64Array(6) [ 2, 4, 3, 6, 5, NaN ]
```

With Fortran

```fortran
subroutine dlaswp(n, a, lda, k1, k2, ipiv, incx)
      INTEGER            INCX, K1, K2, LDA, N
      INTEGER            IPIV( * )
      DOUBLE PRECISION   A( LDA, * )
      INTEGER            I, I1, I2, INC, IP, IX, IX0, J, K, N32
      DOUBLE PRECISION   TEMP
      IF( incx.GT.0 ) THEN
         ix0 = k1
         i1 = k1
         i2 = k2
         inc = 1
      ELSE IF( incx.LT.0 ) THEN
         ix0 = k1 + ( k1-k2 )*incx
         i1 = k2
         i2 = k1
         inc = -1
      ELSE
         RETURN
      END IF
      n32 = ( n / 32 )*32
      IF( n32.NE.0 ) THEN
         DO 30 j = 1, n32, 32
            ix = ix0
            DO 20 i = i1, i2, inc
               ip = ipiv( ix )
               IF( ip.NE.i ) THEN
                  DO 10 k = j, j + 31
                     temp = a( i, k )
                     a( i, k ) = a( ip, k )
                     a( ip, k ) = temp
   10             CONTINUE
               END IF
               ix = ix + incx
   20       CONTINUE
   30    CONTINUE
      END IF
      IF( n32.NE.n ) THEN
         n32 = n32 + 1
         ix = ix0
         DO 50 i = i1, i2, inc
            ip = ipiv( ix )
            IF( ip.NE.i ) THEN
               DO 40 k = n32, n
                  temp = a( i, k )
                  a( i, k ) = a( ip, k )
                  a( ip, k ) = temp
   40          CONTINUE
            END IF
            ix = ix + incx
   50    CONTINUE
      END IF
   RETURN
end subroutine

PROGRAM test_dlaswp
   INTEGER :: n, lda, k1, k2, incx
   INTEGER, ALLOCATABLE :: ipiv(:)
   DOUBLE PRECISION, ALLOCATABLE :: a(:,:)

   ! Define the input values
   n = 3
   lda = 2
   k1 = 1
   k2 = 3
   incx = 1

   ! Allocate the arrays
   ALLOCATE(ipiv(n))
   ALLOCATE(a(lda, n))

    ! Initialize the arrays
   a = RESHAPE((/ 1.0D0, 2.0D0, 3.0D0, 4.0D0, 5.0D0, 6.0D0 /), SHAPE(a))

    ! Initialize the pivot array
    ipiv = (/ 3, 1, 2 /)

   ! Print the original matrix
   PRINT *, "Original matrix A:"
   DO i = 1, lda
       PRINT *, a(i, :)
   END DO

   ! Call the subroutine
   CALL dlaswp(n, a, lda, k1, k2, ipiv, incx)

   ! Print the modified matrix
   PRINT *, "Modified matrix A:"
   DO i = 1, lda
       PRINT *, a(i, :)
   END DO
END PROGRAM test_dlaswp
```

```console
% lfortran b.f90 --implicit-typing
Original matrix A:
1.00000000000000000e+00 3.00000000000000000e+00 5.00000000000000000e+00 
2.00000000000000000e+00 4.00000000000000000e+00 6.00000000000000000e+00 
Modified matrix A:
2.00000000000000000e+00 3.00000000000000000e+00 5.00000000000000000e+00 
4.00000000000000000e+00 6.00000000000000000e+00 1.00000000000000000e+00 
(lf) pranavchiku@pranavchikus-MacBook-Air:~/repos/stdlib-js/stdlib(dlaswp) % gfortran b.f90 && ./a.out
 Original matrix A:
   1.0000000000000000        3.0000000000000000        5.0000000000000000     
   2.0000000000000000        4.0000000000000000        6.0000000000000000     
 Modified matrix A:
   2.0000000000000000        3.0000000000000000        5.0000000000000000     
   4.0000000000000000        6.0000000000000000        1.0000000000000000   
```

I think something is off, which I am trying to debug.

Reply:
I think we can mark this as resolved, correct?